shtdlib.sh

```diff
@@ -191,11 +191,12 @@ function umask_decorator {
 #     # 'sort' doesn't properly handle SIGPIPE
 #     shopt_decorator_option_name='pipefail'
 #     shopt_decorator_option_value='false'
-#     shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return
+#     shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return ${?}
```

Comment:
Wouldn't this always return 0 since its after a `&&`?

Reply:
Correct, this is a pointless addition, removing.

Reply:
Rewrote error handling for decorators to be better.